homeassistant/components/philips_js/diagnostics.py

```diff
@@ -0,0 +1,54 @@
+"""Diagnostics support for AirVisual."""
+from __future__ import annotations
+
+from typing import Any
+
+from homeassistant.components.diagnostics import async_redact_data
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+
+from . import PhilipsTVDataUpdateCoordinator
+from .const import DOMAIN
+
+TO_REDACT = {"serialnumber_encrypted", "serialnumber", "deviceid_encrypted", "deviceid"}
+
+
+async def async_get_config_entry_diagnostics(
+    hass: HomeAssistant, entry: ConfigEntry
+) -> dict[str, Any]:
+    """Return diagnostics for a config entry."""
+    coordinator: PhilipsTVDataUpdateCoordinator = hass.data[DOMAIN][entry.entry_id]
+    api = coordinator.api
+
+    return {
+        "entry": {
+            "title": entry.title,
+            "data": async_redact_data(entry.data, TO_REDACT),
+            "options": async_redact_data(entry.options, TO_REDACT),
```

Comment:
Why not provide the full config_entry?

Reply:
I meant more in line with

```suggestion
            "config": async_redact_data(config_entry.as_dict(), TO_REDACT)
```